classes/ActionScheduler_wpCommentLogger.php

```diff
@@ -26,7 +26,7 @@ public function log( $action_id, $message, DateTime $date = NULL ) {
 
 	protected function create_wp_comment( $action_id, $message, DateTime $date ) {
 		$comment_date_gmt = $date->format('Y-m-d H:i:s');
-		$date->setTimezone( ActionScheduler_TimezoneHelper::get_local_timezone() );
+		ActionScheduler_TimezoneHelper::set_local_timezone( $date );
```

Comment:
There's a potential issue here:

```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to ActionScheduler_TimezoneHelper::set_local_timezone() must be an instance of ActionScheduler_DateTime, instance of DateTime given, called in /Users/jpry/Sites/prospress/content/plugins/action-scheduler/classes/ActionScheduler_wpCommentLogger.php on line 29 and defined in /Users/jpry/Sites/prospress/content/plugins/action-scheduler/classes/ActionScheduler_TimezoneHelper.php:16
Stack trace:
#0 /Users/jpry/Sites/prospress/content/plugins/action-scheduler/classes/ActionScheduler_wpCommentLogger.php(29): ActionScheduler_TimezoneHelper::set_local_timezone(Object(DateTime))
#1 /Users/jpry/Sites/prospress/content/plugins/action-scheduler/classes/ActionScheduler_wpCommentLogger.php(23): ActionScheduler_wpCommentLogger->create_wp_comment(2989, 'from shell!!!', Object(DateTime))
#2 phar:///usr/local/Cellar/wp-cli/2.0.1/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/REPL.php(37) : eval()'d code(1): ActionScheduler_wpCommentLogger->log(2989, 'from shell!!!', Object(DateTime))
#3 phar in /Users/jpry/Sites/prospress/content/plugins/action-scheduler/classes/ActionScheduler_TimezoneHelper.php on line 16
```

I see this as the main blocking issue, as seen in 5843134. While our own code doesn't natively pass an instantiated object, it's possible other code could be doing this.

I added 3f9017200bb73fa2907239186b5202b1c57aecdd as an attempt to bypass this issue. For this most part, this is simply updating docblocks to describe the current behavior. Can you think of any other areas (besides the custom tables extension) where this might be an issue?